presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -186,24 +185,17 @@ public PlanNode visitExchange(ExchangeNode exchange, RewriteContext<FragmentProp
 
             PartitioningScheme partitioningScheme = exchange.getPartitioningScheme();
 
-            ImmutableList.Builder<SubPlan> builder = ImmutableList.builder();
             if (exchange.getType() == ExchangeNode.Type.GATHER) {
                 context.get().setSingleNodeDistribution();
-
-                for (int i = 0; i < exchange.getSources().size(); i++) {
-                    FragmentProperties childProperties = new FragmentProperties(partitioningScheme.translateOutputLayout(exchange.getInputs().get(i)));
-                    builder.add(buildSubPlan(exchange.getSources().get(i), childProperties, context));
-                }
             }
             else if (exchange.getType() == ExchangeNode.Type.REPARTITION) {
                 context.get().setDistribution(partitioningScheme.getPartitioning().getHandle());
-
-                FragmentProperties childProperties = new FragmentProperties(partitioningScheme.translateOutputLayout(Iterables.getOnlyElement(exchange.getInputs())));
-                builder.add(buildSubPlan(Iterables.getOnlyElement(exchange.getSources()), childProperties, context));
             }
-            else if (exchange.getType() == ExchangeNode.Type.REPLICATE) {
-                FragmentProperties childProperties = new FragmentProperties(partitioningScheme.translateOutputLayout(Iterables.getOnlyElement(exchange.getInputs())));
-                builder.add(buildSubPlan(Iterables.getOnlyElement(exchange.getSources()), childProperties, context));
+
```

Comment:
It seems to be a typo in commit message. It should be `Support multiple sources in REPARTITION and REPLICATE`.

Maybe I was a little bit too eager by adding support for `REPLICATE`, but I didn't find a fundamental problem why it shouldn't work. Each `REPLICATE` `ExchangeOperator` will just read from more sources. Instead of reading from n operator's on n workers, it will pull the data from 2*n operator on n workers.